### ISSUE TITLE
[FR]: Add windows support for CLI versions > 5.5.0

### ISSUE THREAD

**Issue by RobertDiebels** [aspect-build/aspect-cli#598]
### What is the current behavior?

According to the docs:
"Windows releases are no longer published as of [v5.5.0](https://github.com/aspect-build/aspect-cli/releases/tag/5.5.0). This is due to a new dependency on [go-tree-sitter](https://github.com/tree-sitter/tree-sitter-go) which has a CGO dependency that makes cross compiling to Windows difficult."

### Describe the feature

I would like to have Windows support for Aspect CLI. I'm trying to build the [most recent version](https://github.com/aspect-build/rules_ts/releases/tag/v2.1.0) of rules_ts and it fails due to a dependency on Aspect CLI.

I've looked in the [treesitter-go](https://github.com/tree-sitter/tree-sitter-go) repository, but there is no open issue that mentions cross-compilation failure. I realize that this an upstream issue, but if you could at least file an issue with them detailing the issue then they can have a look at it.

**Comment by alexeagle** [comment]
Yeah, this is a problem, we haven't had time to dig into it. A bug bounty would be helpful here, since we don't have many Bazel users on Windows asking for it.

**Comment by RobertDiebels** [comment]
@alexeagle Is there anything I can do to move this along? I'm willing and able to create a PR, I just need a nudge into the right direction, which might undercut the need for funding.

**Comment by alexeagle** [comment]
Sure, the hard part is cross-compiling the Go binary including cgo dependencies.
Without that we could try some GHA release automation changes to build the exe on a Windows runner, though process changes are harder to contribute

**Comment by RobertDiebels** [comment]
@alexeagle Does Aspect cross-compile the Go binary? Or is this done elsewhere? Could you point me to where this occurs so I can take a look at it?

**Comment by alexeagle** [comment]
You can build the //release package, see https://github.com/aspect-build/aspect-cli/blob/main/release/release.bzl#L38

**Comment by RobertDiebels** [comment]
@alexeagle Awesome thank you, do you think it would be worth trying to swap out the llvm toolchain for [the zig-based hermetic_cc_toolchain from Uber](https://github.com/uber/hermetic_cc_toolchain?tab=readme-ov-file#use-case-manually-build-a-single-target-with-a-specific-zig-cc-toolchain)?

Zig isn't as battle-tested as LLVM, but their compiler has CC as a [first-class use case](https://ziglang.org/learn/overview/#cross-compiling-is-a-first-class-use-case) and they released [version 0.12.0](https://ziglang.org/download/0.12.0/release-notes.html#Tier-1-Support ) 2 days ago.

I'd like to check with you before I attempt to swap out LLVM to make sure you'd be willing to adopt zig (either partially just for windows, or fully for all binaries) as not to waste time/effort in an attempt.

**Comment by alexeagle** [comment]
This repo is synced with our internal monorepo so changing the toolchain is likely breaking for something else, and so this is an area where it's hard for us to accept external contribution.
All things equal I think I'm more comfortable relying on GHA windows machines for releases rather than more complexity in our Bazel toolchain. @gregmagolan WDYT?

**Comment by albertocavalcante** [comment]
Hey, just wondering if there's been further discussion on this. I'd like to test Aspect CLI in an environment where my Windows machine does not have WSL2 enabled. I will give it a try on RHEL Linux but anyway Windows would still be desired hence my ask. Thanks.

**Comment by RobertDiebels** [comment]
> Hey, just wondering if there's been further discussion on this. I'd like to test Aspect CLI in an environment where my Windows machine does not have WSL2 enabled. I will give it a try on RHEL Linux but anyway Windows would still be desired hence my ask. Thanks.

@albertocavalcante 

I don't know if there has been any internal discussion, but you could try to swap out the cross-compilation tool chain currently used. I've recently found out that it's already being used when I ran into [another issue](https://github.com/aspect-build/bazel-lib/issues/849) [ https://github.com/aspect-build/bsdtar-prebuilt/blob/main/MODULE.bazel ].

I'd attempt to do it myself. Unfortunately I will be on holiday soon and don't have time to spare leading up to it.

**Comment by mattem** [comment]
We have an [internal diff](https://github.com/aspect-build/silo/pull/5812) to transition to Zig, which will if it lands, will be exported here. + @mrmeku for visibility.

**Comment by mrmeku** [comment]
I've been playing with swapping in the zig toolchain and I've run into compatibility issues with compiling tree sitter. I'm unfortunately out of my depth and need a C++ build oriented developer to proceed further

**Comment by alexeagle** [comment]
Nope, no discussion or work on this since we don't have any paying customers asking for it. PRs welcome! Maybe this is easy to do by adding a windows CD runner on GHA.

**Comment by peakschris** [comment]
There now seem to be windows binaries published on github; but they are not hosted at a location they are picked up by bazelisk (at least for bazel-lib):

`https://static.aspect.build/aspect`

Please could they be hosted there?
